src/evm/engine_db/simulation_db.rs

```diff
@@ -322,11 +314,11 @@ where
     }
 }
 
-impl<M: Middleware + Debug> DatabaseRef for SimulationDB<M>
+impl<P: Provider + Debug> DatabaseRef for SimulationDB<P>
 where
-    M::Error: std::fmt::Debug,
+    P: Provider + Send + Sync + 'static,
 {
-    type Error = M::Error;
+    type Error = Box<dyn std::error::Error + Send + Sync>;
```

Comment:
Can't we make this not `dyn` but a known type instead?

Reply:
huumm the type depends on the provider I think. that's why I didn't want to define it here